src/components/Upload.jsx

```diff
@@ -46,10 +46,10 @@ const Upload = ({ setCurrentScript, titles, setTitles }) => {
         }
         return response.json();
       })
-      .then((title) => {
+      .then((script) => {
         // set the current script to the newly uploaded script
-        setCurrentScript(title);
-        setTitles([...titles, title]);
+        setCurrentScript(script.id);
+        setTitles([...titles, script]);
```

Comment:
titles is now an array of objects:
```js
{id, title}
```

Reply:
This was also updated to be `scripts` and `setScripts` because titles is misleading after these changes